environments/gcp/gke.tf

```
@@ -0,0 +1,34 @@
+locals {
+  gke_subnetwork          = module.network.subnets_names[0]
+  gke_pods_range_name     = module.network.subnets_secondary_ranges[0][0].range_name
+  gke_services_range_name = module.network.subnets_secondary_ranges[0][1].range_name
+}
+
+module "gke" {
+  source                   = "terraform-google-modules/kubernetes-engine/google"
+  project_id               = local.project_id
+  region                   = local.region
+  name                     = local.name_prefix
+  regional                 = true
+  release_channel          = "STABLE"
+  network                  = module.network.network_name
+  subnetwork               = local.gke_subnetwork
+  ip_range_pods            = local.gke_pods_range_name
+  ip_range_services        = local.gke_services_range_name
+  create_service_account   = true
+  remove_default_node_pool = true
+
+  node_pools = [
+    {
+      name         = "default"
+      machine_type = "e2-standard-4"
+      min_count    = 0
+      max_count    = 3
```

Suggestion

```suggestion
      max_count    = 3
      # Activate image streaming https://cloud.google.com/kubernetes-engine/docs/how-to/image-streaming to speed up pulling of images
      enable_gcfs  = false  
```

Let's not activate this by default (default is false) but let's point users to this feature, it is amazing. Our images are several GB large typically and instead of pulling them for several minutes, it takes <10 seconds. This makes for a super smooth Flyte experience.